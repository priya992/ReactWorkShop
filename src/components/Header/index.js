import React, {useState, useEffect} from 'react';
import useDebounce from '../../hooks/useDebounce';

const Header = (props) => {
  const {pageTitle, getSearchData, closeSearch} = props

  const [displaySearch, setDisplaySearch] = useState(false)
  const [searchValue, setSearchValue] = useState(null)

  const debouncedSearchTerm = useDebounce(searchValue, 100);

  useEffect(
   () => {
      getSearchData(debouncedSearchTerm)
   },
   [debouncedSearchTerm]
 );

  useEffect(
   () => {
     if (!displaySearch) {
       closeSearch()
     }
   },
   [displaySearch]
 );

  const displaySearchHandler = () => {
    setDisplaySearch(!displaySearch)
  }

  const close = () => {
    setSearchValue(null)
    setDisplaySearch(!displaySearch)
  }

  const searchHandler = (event) => {
    const value = event.target.value
    setSearchValue(value)
  }

  const withoutSearch = (
    <div className="header-container">
      <div className="header-sub">
        <img src="media/Back.png" alt="back-btn"/>
        <div className="header-text">{pageTitle}</div>
      </div>
      <img src="media/search.png" alt="search" onClick={displaySearchHandler}/>
    </div>
  )

  const withSearch = (
    <div className="header-container">
      <img src="media/Back.png" alt="back-btn"/>
      <input
         className="search-input"
         placeholder="Search for..."
         value={searchValue}
         onChange={searchHandler}
       />
       <img src={"media/close.svg"} alt="close-btn" className="search-image" onClick={close}/>
    </div>
  )

  if (displaySearch) {
    return withSearch
  }

  return withoutSearch
}

export default Header;
