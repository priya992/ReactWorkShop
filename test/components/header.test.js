import React from 'react';
import renderer from 'react-test-renderer';

import Header from './../../src/components/Header';
import { headerProps } from './../__mock__';

test('Header component snapshot', () => {
  const component = renderer.create(
    <Header {...headerProps}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
