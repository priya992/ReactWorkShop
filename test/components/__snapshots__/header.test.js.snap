// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component 1`] = `
<div
  className="header-container"
>
  <div
    className="header-sub"
  >
    <img
      alt="back-btn"
      src="media/Back.png"
    />
    <div
      className="header-text"
    >
      Romantic Comedy
    </div>
  </div>
  <img
    alt="search"
    onClick={[Function]}
    src="media/search.png"
  />
</div>
`;
